Bookish/Controllers/CatalogController.cs

```diff
@@ -49,41 +59,161 @@ public IActionResult BookEntry(BookInputModel model)
         {
-            bvm.EntryQuery = context.Books.Include(b => b.Author)
-                .Include(b => b.CopyList);
-            bvm.CatalogEntries = bvm.EntryQuery.ToList();
+            bvm.CatalogEntries = context.Books.Include(b => b.Author)
+                .Include(b => b.CopyList)
+                .ToList();
         }
-        return View(bvm);
+        return View("BookList",bvm);
     }
+    /*[HttpPost]
+    public IActionResult BookList(int bookId, string operation)
+    {
+        switch (operation)
+        {
+            case "delete":
+                throw new Exception("HELP DELETE");
+                return Delete(new BookViewModel());
+                break;
+            case "edit":
+                throw new Exception("HELP EDIT");
+                return Edit(new BookViewModel());
+                break;
+            default:
+                return View("Error");
+        }
+    }*/
+    [HttpGet]
+    public IActionResult Delete(int bookId)
+    {
 
+        Book? foundBook;
+        using (var context = new LibraryContext())
+        {
+            foundBook = context.Books.SingleOrDefault(a => a.BookID == bookId );
+            if (foundBook == null)
+            {
+                throw new Exception("NO BOOK!");
+            }
+            else
+            {
+                context.Remove(foundBook);
+                context.SaveChanges();
+            }
+        }
+
+        return BookList();
+    }
+    [HttpGet]
+    public IActionResult Edit(int bookId)
+    {
+        Book? foundBook;
+        using (var context = new LibraryContext())
+        {
+            foundBook = context.Books.SingleOrDefault(a => a.BookID == bookId);
+            foundBook.Author = context.Authors.SingleOrDefault(a => a.AuthorID == foundBook.AuthorID);
+            if (foundBook == null)
+            {
+                throw new Exception("NO BOOK!");
+            }
+        }
+        return View(foundBook);
+    }
+
+    [HttpPost]
+    public IActionResult Edit(Book b)
+    {
+        using (var context = new LibraryContext())
+        {
+            Book? foundBook = context.Books.SingleOrDefault(a => a.BookID == b.BookID);
+            Author? foundAuthor = context.Authors.SingleOrDefault(a =>
+                a.AuthorForename == b.Author.AuthorForename && a.AuthorSurname == b.Author.AuthorSurname);
+            if (foundBook == null)
+            {
+                //error stuff
+            }
+            else
+            {
+                if (foundAuthor == null)
+                {
+                    foundAuthor = new Author()
+                        { AuthorForename = b.Author.AuthorForename, AuthorSurname = b.Author.AuthorSurname };
+                    
+                    context.Authors.Add(foundAuthor);
+                    context.SaveChanges();
+                }
+                b.Author = foundAuthor;
+                Book _b = context.Books.Where(s => s.BookID == b.BookID).FirstOrDefault<Book>();

```

Comment:
`_b` isn't a very friendly variable name - need to read the actual logic of the code to get an idea of what this variable is being used for